### DISCUSSION TITLE
Keybind to close assistant panel when it is in focus

### DISCUSSION THREAD

**Discussion by zakajus** [zed-industries/zed#13689] [Feature Requests]
Referencing #13342, I have a keybind to close the project panel with escape if it is focused, which works perfectly.
```
{
        "context": "ProjectPanel && not_editing",
        "bindings": {
            "escape": "workspace::ToggleLeftDock"
        }
}
```

I want the same for the assistant panel. I have tried the following, however it does not work:
```
{
        "context": "AssistantPanel && not_editing",
        "bindings": {
            "escape": "workspace::ToggleRightDock"
        }
}
```
What is the proper way to do this?

Thank you!

**Comment by tepavcevic** [comment]
It doesn't look like there is a straightforward solution for this, it seems it's not yet implemented with the default keymap. But what you could do is to do something like this:

```json
{
    "context": "Editor (and whatever additional context depending if you use vim mode)",
    "bindings": {
        "escape escape (or whatever makes sense to you)": "workspace::ToggleRightDock"
    }
}
```
Since I don't use assistant, I wouldn't know if this works when it's enabled, so you might need to add the same binding for AssistantPanel, but it works for me as is.
Until there is a better solution, I think this could work

**Comment by kualta** [comment]
this should be an issue instead, AssistantPanel is ignored in the context selection it feels like

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
